### PR TITLE
blockresolver

### PR DESCRIPTION
**Pull request by tom-wa**
```
% cat /tmp/test.block
bla
bla
bla
### block config start
[section1]
key1 = val1
[section2]
key2 = val2
### block config stop
asdf
asdf
asdf

% kdb mount -R blockresolver /tmp/test.block system/blocktest -c identifier="### block config"
% kdb export system/blocktest
[section1]
key1 = val1
[section2]
key2 = val2

% kdb set system/blocktest/section1/key12 val12
Create a new key system/blocktest/section1/key12 with string val12

% cat /tmp/test.block
bla
bla
bla
### block config start
[section1]
key1 = val1
key12 = val12
[section2]
key2 = val2
### block config stop
asdf
asdf
asdf
```

### CODE REVIEW

File: src/plugins/blockresolver/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+Copy this blockresolver if you want to start a new
+plugin written in C.
+
+You can use scripts/copy-blockresolver
+to automatically rename everything to your
+plugin name:
+
+	cd src/plugins
+	../../scripts/copy-blockresolver yourplugin
+
```
Comment (markus2330) [inline src/plugins/blockresolver/README.md]: please put the description of commit also here.

**Comment by markus2330** [comment]
Thank you! Your description looks like it does what it should!

File: src/plugins/blockresolver/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+- infos = Information about the blockresolver plugin is in keys below
+- infos/author = Name <name@libelektra.org>
+- infos/licence = BSD
+- infos/needs =
+- infos/provides = resolver
+- infos/recommends =
+- infos/placements = rollback getresolver setresolver commit
+- infos/status = recommended productive maintained reviewed conformant compatible coverage specific unittest shelltest tested nodep libc configurable final preview memleak experimental difficult unfinished old nodoc concept
```
Comment (markus2330) [inline src/plugins/blockresolver/README.md]: please keep up-to-date

**Comment by markus2330** [comment]
What about chaining the resolvers to avoid duplication of update/commit logic?
In this concrete case I am not sure though. E.g. the empty file handling might be wrong, because even with empty block you still have the surrounding file. (So chaining might cause more problems than it helps.)

But for multi-file resolvers it is definitely worth a thought.

**Comment by markus2330** [comment]
Some tests are failing here.

**Comment by markus2330** [comment]
Some tests are still failing.

**Comment by tom-wa** [comment]
seems like i forgot to push the testfiles too...

**Comment by markus2330** [comment]
@BernhardDenner Do you want to take a look?

**Comment by markus2330** [comment]
@BernhardDenner ping

@tom-wa please rebase.

**Comment by markus2330** [comment]
Thank you!

**Pull request merged**